rest_framework_simplify/serializer.py

```diff
@@ -154,7 +154,10 @@ def model_to_dict(self, obj):
             json_data_str = serializers.serialize('json', [obj])
             plain_dict = json.loads(json_data_str)
             model_dict = plain_dict[0]['fields']
-            model_dict['id'] = plain_dict[0]['pk']
+            if obj._meta.pk.attname != 'id':
```

Comment:
This is good the only thing that could bite us in the ass is if the obj doesn't have a _meta attribute or the _meta doesnt have a pk or the pk doesnt have an attname. But I think we can be ok with this.